spec/models/trainee_spec.rb

```diff
@@ -593,6 +593,49 @@ def subject.serializable_hash(*args)
     end
   end
 
+  describe "#course_programme_year" do
+    it "returns nil if no itt start/end dates set" do
+      trainee = Trainee.new(itt_start_date: Time.zone.today, itt_end_date: nil)
+      expect(trainee.course_programme_year).to be_nil
+      trainee = Trainee.new(itt_start_date: nil, itt_end_date: Time.zone.today)
+      expect(trainee.course_programme_year).to be_nil
+    end
+
+    describe "returns 1 year course" do
+      it "returns 1 within course period" do
+        trainee = Trainee.new(itt_start_date: 3.months.ago.to_date, itt_end_date: 3.months.from_now.to_date)
+        expect(trainee.course_duration_in_years).to eq(1)
+        expect(trainee.course_programme_year).to eq(1)
+      end
+
+      it "returns 1 after course period" do
+        trainee = Trainee.new(itt_start_date: 12.months.ago.to_date, itt_end_date: 6.months.ago.to_date)
+        expect(trainee.course_duration_in_years).to eq(1)
+        expect(trainee.course_programme_year).to eq(1)
+      end
+    end
+
+    describe "returns 2 years course" do
+      it "returns 1 within course 1st year period" do
+        trainee = Trainee.new(itt_start_date: 3.months.ago.to_date, itt_end_date: 18.months.from_now.to_date)
+        expect(trainee.course_duration_in_years).to eq(2)
+        expect(trainee.course_programme_year).to eq(1)
```

Comment:
Are we saying that the programme year is 1 in a 2 year course because they haven't started year 2 yet? If DTTP was going to stay on, would this mean that when the trainee enters year 2, then another update would have to be sent to DTTP with the change in programme year?